src/contentBase.ts

```diff
@@ -88,23 +88,44 @@ export abstract class ContentBase implements IPackageMetadata {
             return null;
         }
         searchFile = searchFile.toLowerCase();
+        let foundPath = null;
         for(const filePath of fileList) {
             if(filePath.toLowerCase() === searchFile) {
                 return filePath;
             }
-            const pathSplit =  filePath.split(path.sep);
+            const pathSplit = filePath.split(path.sep);
             for(const pathSegment of pathSplit) {
                 if(pathSegment.toLowerCase() === searchFile && (searchFile !== Constants.INFO_PLIST || searchFile === Constants.PROVISIONING)) {
-                    return filePath;
+                    foundPath = this.selectShortestPath(foundPath, filePath);
                 }
                 // Info.plist and embedded.mobileprovisions is a special case since there are multiple instances in different files
                 if(pathSegment.toLowerCase() === searchFile && (searchFile === Constants.INFO_PLIST || searchFile === Constants.PROVISIONING) && pathSplit.length <= 3) {
-                    return filePath;
+                    foundPath = this.selectShortestPath(foundPath, filePath);
                 }
             }
         }
-        return null;
+        return foundPath;
     }
+
+
+    /**
+     * This method determines which path (shortest one) should be selected based on its file path depth length. 
```

Comment:
How about:
```
This method compares 2 paths and returns the one with shortest path to a file. 
For example:
path1: a/b/c.txt
path2: a/c.txt

will return path2 since it has the shortest path to c.txt
```